### PR TITLE
chore: Generate between split repos and main

### PR DESCRIPTION
**Pull request by lqiu96**
Following Tomo's script advice:
```
#!/bin/bash

set -e

modules=$(mvn help:evaluate -Dexpression=project.modules | grep '<.*>.*</.*>' | grep 'java' | sed -e 's/<.*>\(.*\)<\/.*>/\1/g')

for module in $modules; do
  echo "Running for ${module}"
  rm -rf "${module}"
  git clone "https://github.com/googleapis/${module}"
  rm -rf "${module}/.git"
  echo "Done running for ${module}"
done

./generation/delete_non_generated_samples.sh

echo "Done running script"
```

For now:
- Ignore pom.xml (assuming they are renovate PRs) -- Can create PR to check only `pom.xml`s
- Ignore CHANGELOG.md (from release-please) -- Might need to copy over from split-repos
- Ignore README.md -- Can copy over from split repos
- Ignore split repos renovate.json configs 
- Ignore samples/*
`git diff --name-only origin/main | grep -v "pom.xml" | grep -v "CHANGELOG.md" | grep -v "README.md" | grep -v "renovate.json" | grep -v "versions.txt" | grep -v "samples/*" > diff.txt`

Probably should ignore owlbot files and .github/*

Shows only the name of the files that have changed: [diff.txt](https://github.com/googleapis/google-cloud-java/files/9630498/diff.txt)

Github has issue displaying the diffs. Might need to only add the files listed in the diff.txt

### CODE REVIEW

**Comment by suztomo** [comment]
This PR reveals Java code diff (maybe manually-maintained tests?). Great.

Screen Shot 2022-09-23 at 11 22 41 AM

**Comment by lqiu96** [comment]
> How about limiting first 1000 diff files in the pull request? (We can't see thousands of files anyway)

I didn't want to manually limit/exclude certain files. I've taken a look at a few of the diffs locally and I think quite a few of them can be ignored.

Ignoring these files should help us limit the number of diffs we need to check:
- CHANGELOG.md (from release-please) -- Might need to copy over from split-repos
- README.md -- Can copy over from split repos
- renovate.json configs in each split repos
- Ignore samples/* -- As we are no longer maintaining them and can look into them later
- Ignore owlbot.yamls

Need to check:
- The .java files (either manually or from owlbot)

The pom.xml files shouldn't have custom pom.xml, but I can do a separate and specific diff to check each pom.xml. What do you think about the proposed ignored files above? @suztomo 

> Once the script is ready and merged to main, we want a GitHub Actions job that updates a branch periodically. We monitor a pull request from the branch to observe the changes.

I can look into a GH action once I flush out the script with the diffs that I want to show.

**Comment by lqiu96** [comment]
Plan forward is two branches
1. Java only code (from owlbot changes and samples)
2. Non-Java code

Script will create two branches and create PRs for each branch that show the diffs

**Comment by suztomo** [comment]
We need 1 script generation/generate_diff.sh.


# 1st GitHub Actions job

```
steps:
- run: |
  generation/generate_diff.sh
- run: |
  # Commit only Java code.
  git commit ./**/*.java -m 'chore: Java code diff from split repos'
- run: |
  git push -f --set-upstream origin split-repo-diff-non-java
```

# 2nd GitHub Actions job

```
steps:
- run: |
  generation/generate_diff.sh
- run: |
  # Commit non-Java code. How??
  git commit  ???????? -m 'chore: Java code diff from split repos'
- run: |
  git push -f --set-upstream origin split-repo-diff-non-java
```

**Review by suztomo — DISMISSED** [review]

File: generation/generate_diff.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+git add "*.java"
+git commit -m "chore: Adding java diffs" --no-verify
+git push origin "${diff_java_branch}" --force
+
+git stash
+git checkout "${current_branch}"
+git stash pop
+
+git checkout -b "${diff_non_java_branch}"
+git add .
```
Comment (suztomo) [inline generation/generate_diff.sh:85]: I like the use of `git stash` to separete Java change and non-Java changes.

File: .github/workflows/generate_diffs.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  generate-diffs:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        repository: 'googleapis/google-cloud-java'
+    - name: Run generate-diff script
+      run: ./generation/generate_diff.sh
+      env:
+        USERNAME: ${{ github.actor }}
```
Comment (suztomo) [inline .github/workflows/generate_diffs.yaml:37]: Did it work? If not `YOSHI_CODE_BOT_TOKEN` is available.

https://github.com/googleapis/google-cloud-java/blob/main/.github/workflows/googleapis_hermetic_sync.yaml#L41

File: generation/generate_diff.sh (hunk 1 above)
Comment (lqiu96) [reply generation/generate_diff.sh:85]: Let me know if you have a better option. I couldn't think besides this that wouldn't require either two jobs or running the script twice.

File: .github/workflows/generate_diffs.yaml (hunk 2 above)
Comment (lqiu96) [reply .github/workflows/generate_diffs.yaml:37]: Yep, https://github.com/googleapis/google-cloud-java/pull/8448 and https://github.com/googleapis/google-cloud-java/pull/8449 both have `github-actions force-pushed...`

File: .github/workflows/generate_diffs.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
```
Comment (suztomo) [inline .github/workflows/generate_diffs.yaml:19]: I assume you're testing. We don't need this job at pull request checks.

Comment (lqiu96) [reply .github/workflows/generate_diffs.yaml:19]: Yep! Just added in https://github.com/googleapis/google-cloud-java/pull/8442/commits/fa84ab5974d6b88d45457cd243c4212e5ea78454 to run on non-PR `push` events

Comment (suztomo) [reply .github/workflows/generate_diffs.yaml:19]: It appears in the checks. Can you fix this?

Screen Shot 2022-09-23 at 4 49 29 PM

Comment (lqiu96) [reply .github/workflows/generate_diffs.yaml:19]: Hmm, I'm sure if I can prevent it from showing up on the checks. It seems to show up on a `push` event. 

I can skip it from running all together.
image

Comment (suztomo) [reply .github/workflows/generate_diffs.yaml:19]: Would you explain why you need this line 19 "  pull_request:" ?

Comment (suztomo) [reply .github/workflows/generate_diffs.yaml:19]: > I can skip it from running all together.

Yes, please do so. Remove line 19.

Comment (lqiu96) [reply .github/workflows/generate_diffs.yaml]: Ah good catch! I've removed the check.

**Review by suztomo — APPROVED** [review]

**Comment by lqiu96** [comment]
I'll merge this now to have it regenerate the diffs on merge. I'll create a separate PR to split the ITs and other Java code.

**Pull request merged**